torch/csrc/lazy/core/dynamic_ir.cpp

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <torch/csrc/lazy/core/dynamic_ir.h>
+
+namespace torch {
+namespace lazy {
+
+DimensionNode::DimensionNode(OpKind op, OpList operands, hash_t hash_seed):
+  Node(op, /*num_outputs=*/1, 
```

Comment:
I'm surprised `Node` class is bare-bones. @wconstab do you know why we don't even keep `operands_` in Node?

Reply:
i think we could move it back, at one point i thought we might not want it/need it for TS backend, but that didn't pan out